Sources/Client/Client/ClientLogger.swift

```diff
@@ -152,19 +156,26 @@ public final class ClientLogger {
     Customizing the logger block with `ClientLogger.logger` is deprecated.
     Please use `ClientLogger.log` to customize your logging logic.
     """)
-    public static var logger: (String, String, String) -> Void = { icon, dateTime, message in
+    public static var logger: (String, String, String) -> Void = legacyLogger {
```

Comment:
Maybe better to make it optional and we won't need `usingLegacyLogger` flag.

Reply:
How?

Reply:
```suggestion
    public static var logger: ((String, String, String) -> Void)?
```
and then
```
         if let legacyLogger = ClientLogger.logger {
             legacyLogger(icon, dateTime, message)
         } else {
             ClientLogger.log(icon, dateTime, level, message)
         }
```

Reply:
You'll see deprecation warning since you're accessing `ClientLogger.logger`, so this won't work

Reply:
You can show a warning in `didSet`

Reply:
How?

Reply:
```suggestion
    // deprecation
    public static var logger: ((String, String, String) -> Void)? {
        didSet {
            // show warning
        }
    }
```

Reply:
@buh that's not possible, at least not on compiler level. We can only show warning in logs with this approach, which is suboptimal.
Please try your suggestions locally before suggesting, so we can save some discussion time.

Reply:
I checked before suggesting.
This is for the current implementation:
```
    public static var logger: (_ icon: String, _ dateTime: String, _ message: String) -> Void = {
        if $1.isEmpty || DateFormatter.log == nil {
            print($0, $2)
        } else {
            print($0, "[\($1)]", $2)
        }
        } {
        didSet {
            print("Check")
        }
    }
```
in usage:
```
        ClientLogger.logger = { _, _, message in
            print("LOG", message)
        }
```
I can see `Check` in logs.

Reply:
@buh We want the warning during compile time, shown in Xcode, not in runtime in console.

Reply:
What do you mean exactly about warning at compile time? It will be the same deprecation warning as you have now. I only suggest that you can avoid `legacyLogger`.

Reply:
@buh This:
```swift
 @available(*, deprecated, message:
```
is a compile time warning. Xcode will show this during build.
You can't put this in `didSet`
If you put this for `logger`, it'll show when `logger` is accessed, so you can't access logger. That's why I shadowed `logger` with `legacyLogger`

Reply:
Ok, I got it.